mathics/core/expression.py

```diff
@@ -905,6 +956,20 @@ def get_sort_key(self, pattern_sort=False):
             else:
                 return [1 if self.is_numeric() else 2, 3, head, self._elements, 1]
 
+    @property
+    def is_literal(self) -> bool:
+        """
+        True if the value can't change, i.e. a value is set and it does not
+        depend on definition bindings. That is why, in contrast to
+        `is_uncertain_final_definitions()` we don't need a `definitions`
+        parameter.
+        """
+        # Right now we are pessimisitic. We might consider changing this for
+        # Lists. Lists definitions can't be changed right?
```

Comment:
A list of numbers shouldn't change. For this, and other structures (like sparse arrays, associations, or subexpressions) I would like to consider using ``InstanceableBuiltin`` specific classes (@rocky, I know, you hate this name, but maybe we can find a better one...) or maybe classes that implement a part of the interface of ``Expression`` but that stores elements in a more suitable way for the specific propose.

Reply:
What I more dislike is talking about mechanisms ("consider using some particular class) rather than what we are trying to accomplish.

The specific question here. is about whether the function "List" function is allowed to be redefined it the meaning of in an S-expression such as `List[1, 2]` can vary.  I get that _after_ evaluation we may want a better or more efficient _representation_ of the result, but that was not the question here.